### PR TITLE
TGP-2302: Fix flaky tests

### PR DESCRIPTION
**Pull request by Reece-Carruthers**
Fix flaky tests by updating generators to prevent non-readable unicode characters being produced, as well as, handling spaces on the end of string generators to ensure correct length. Finally, ensure key generator in RichJsValueSpec is unique to prevent fails due to duplicate data

# Checklist for developers before reviewing
 - [x] I have merged in main
 - [x] I have run  `sbt clean coverage test it/test coverageReport`
 - [ ] I have written unit tests to cover any code that I wrote
 - [x] I have run `sbt scalafmt` and `sbt test:scalafmt`
 - [x] I have updated the microservices
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests
 - [x] I have moved the ticket to Review in Jira
 - [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review

# Checklist for developers before merging
 - [x] The PR has been approved by 2 other developers
 - [x] I have merged in main
 - [x] I have run  `sbt clean coverage test it/test coverageReport`
 - [x] I have run `sbt scalafmtCheckAll`
 - [x] I have updated the microservices
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests
 - [x] I have spoken to the QAs about any changes that may need to be made to the journey tests OR no changes to the journey tests are necessary

# Reminders for developers after merging
 1. Wait for pipeline to pass
 2. Move ticket to QA column in Jira
 3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

File: README.md
Comment (besscerule) [inline README.md]: this does more than just check code coverage it also runs all the unit tests and the integration tests - could you update the name and description to make this clear?

**Review by pauloncast — DISMISSED** [review]

File: build.sbt
Comment (besscerule) [inline build.sbt]: if the scalafmtCheckAll fails will the tests still run and pass? If so it might be difficult to see that scalafmt has failed (similar for the one above, but it is less likely for those to fail as they are doping the actual formatting)

File: pull_request_template.md
Comment (besscerule) [inline pull_request_template.md]: can we include that the tests are run as well?

**Review by besscerule — COMMENTED** [review]
Once this has been merged can you make the same changes to the data-store pr checklist? - and check all the new sbt commands run as expected on there?

File: build.sbt
Comment (Reece-Carruthers) [reply build.sbt]: Yep, so I've just tested there and if the check fails the command will stop which is nice

File: test-utils/generators/Generators.scala
Comment (Ewan-James-Donovan) [inline test-utils/generators/Generators.scala:28]: love this approach

**Review by Ewan-James-Donovan — COMMENTED** [review]
Approving, if there's another commit after then, give me a nudge on slack and I'm happy to re-approve if needed

**Review by Ewan-James-Donovan — DISMISSED** [review]

**Review by besscerule — APPROVED** [review]

**Review by Ewan-James-Donovan — APPROVED** [review]

**Pull request merged**
